core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadOnlyLDAPUserStoreManager.java

```diff
@@ -451,22 +452,35 @@ protected List<String> doGetUserListFromProperties(List<LoginIdentifier> loginId
                             }
                         }
                         String propertyValue = attrBuffer.toString();
+                        // Length needs to be more than userAttributeSeparator.length() for a valid
+                        // attribute, since we
+                        // attach userAttributeSeparator.
+                        if (propertyValue == null || propertyValue.trim().length() <= userAttributeSeparator.length()) {
```

Comment:
Shall we use isNotBlank for strings instead of null

Reply:
Fixed